### PR TITLE
US5858: EdgeX DevOps: figure out how to lint groovy code

### PR DESCRIPTION
**Pull request by vishwassp-hcl-intel**
Signed-off-by: Vishwas SP <vishwasx.s.premachandra@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Sandbox Testing
Test Links :

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -66,20 +68,10 @@ pipeline {
...
             when { expression { edgex.isReleaseStream() } }
             steps {
-                edgeXSemver('bump patch') //this changes the VERSION env var
+                edgeXSemver('bump patch')
                 edgeXSemver('push')
             }
         }
-
-        // automatically bump experimental tag...more research needed
-        stage('🧪 Bump Experimental Tag') {
```
Comment (ernestojeda) [inline Jenkinsfile]: Why are you removing this code??? This is needed to automatically bump the experimental tag

File: Jenkinsfile [hunk 2]
```diff
@@ -34,10 +34,12 @@ pipeline {
             steps {
                 edgeXSetupEnvironment()
                 edgeXSemver 'init'
-                script {
```
Comment (ernestojeda) [inline Jenkinsfile]: Why are you removing this code??? This is needed to automatically bump the experimental tag

File: scripts/linter.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#! /bin/bash
+
+JENKINS_URL=https://jenkins.edgexfoundry.org
+JENKINS_CRUMB=`curl --silent "$JENKINS_URL/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,\":\",//crumb)"`
+JENKINS_VAL="$JENKINS_URL/pipeline-model-converter/validate"
+JENKINS_FILE_LIST=(`grep "pipeline" vars/* | grep -v \/\/ | awk '{print $1}' | sed -e 's/\://g'`)
```
Comment (ernestojeda) [inline scripts/linter.sh]: I would make the first grep be: pipeline {

File: Jenkinsfile (hunk 1 above)
Comment (vishwassp-hcl-intel) [reply Jenkinsfile]: I didn't remove this, let me check

File: scripts/linter.sh (hunk 3 above)
Comment (vishwassp-hcl-intel) [reply scripts/linter.sh]: I didn't do this because, user can use any number of space between pipeline and {

File: Jenkinsfile (hunk 1 above)
Comment (vishwassp-hcl-intel) [reply Jenkinsfile]: Fixed

File: Jenkinsfile (hunk 2 above)
Comment (vishwassp-hcl-intel) [reply Jenkinsfile]: Fixed

File: scripts/linter.sh (hunk 3 above)
Comment (ernestojeda) [reply scripts/linter.sh]: We should be able to use a regular expression to help with that

Comment (vishwassp-hcl-intel) [reply scripts/linter.sh]: OK Done

**Pull request merged**